src/modules/user/user.usecase.ts

```diff
@@ -1377,6 +1379,46 @@ export class UserUseCases {
     );
   }
 
+  async upsertAvatar(
+    user: User,
+    avatarKey: string,
+  ): Promise<Error | { avatar: string }> {
+    if (user.avatar) {
+      try {
+        await this.avatarService.deleteAvatar(user.avatar);
+      } catch (err) {
+        throw new InternalServerErrorException('Failed to delete old avatar');
```

Comment:
We are losing the original error here. The trick would be to keep the message, the stack, etc while throwing the InternalServerErrorException to prevent the error to be leaked to the requester.